k2/csrc/top_sort_test.cu

```diff
@@ -106,11 +106,91 @@ TEST(TopSort, VectorOfFsas) {
 }
 
 TEST(TopSort, RandomSingleFsa) {
-  // TODO(fangjun)
+  for (auto &context : {GetCpuContext(), GetCudaContext()}) {
+    Fsa fsa = GetRandFsa();
+    fsa = fsa.To(context);
+
+    int32_t gt = kFsaPropertiesTopSorted | kFsaPropertiesTopSortedAndAcyclic;
+    int32_t p = GetFsaBasicProperties(fsa);
+    EXPECT_NE(p & gt, gt);
+
+    Fsa sorted;
+    Array1<int32_t> arc_map;
+    TopSort(fsa, &sorted, &arc_map);
+
+    p = GetFsaBasicProperties(sorted);
+    EXPECT_EQ(p & gt, gt);
+
+    Array1<Arc> arcs = sorted.values.To(GetCpuContext());
+    arc_map = arc_map.To(GetCpuContext());
+    int32_t num_arcs = arcs.Dim();
+    for (int32_t i = 0; i != num_arcs; ++i) {
+      EXPECT_EQ(arcs[i].score, arc_map[i]);
+    }
+  }
 }
 
 TEST(TopSort, RandomVectorOfFsas) {
-  // TODO(fangjun)
+  int num_fsas = 1 + rand() % 100;
+  ContextPtr cpu = GetCpuContext();
+  for (auto &context : {GetCpuContext(), GetCudaContext()}) {
+    std::vector<Fsa> fsas(num_fsas);
+    for (int32_t i = 0; i != num_fsas; ++i) {
+      fsas[i] = GetRandFsa();
+    }
+
+    int32_t offset = fsas[0].TotSize(1);
+    for (int32_t i = 1; i != num_fsas; ++i) {
+      Array1<Arc> &arcs = fsas[i].values;
+      Arc *arcs_data = arcs.Data();
+      int32_t num_arcs = arcs.Dim();
+      EXPECT_GT(num_arcs, 1);
+      for (int32_t k = 0; k != num_arcs; ++k) {
+        arcs_data[k].score += offset;
+      }
+      offset += num_arcs;
+    }
+
+    std::vector<Fsa *> fsa_array(num_fsas);
+    for (int32_t i = 0; i != num_fsas; ++i) {
+      fsa_array[i] = &fsas[i];
+    }
+
+    FsaVec fsa_vec = CreateFsaVec(num_fsas, &fsa_array[0]);
+    fsa_vec = fsa_vec.To(context);
+
+    int32_t gt = kFsaPropertiesTopSorted | kFsaPropertiesTopSortedAndAcyclic;
+    Array1<int32_t> properties;
+    int32_t p;
+    GetFsaVecBasicProperties(fsa_vec, &properties, &p);
+
+    /*
+    EXPECT_NE(p & gt, gt);
```

Comment:
Fangjun, was removing these lines OK?  I assume we can't guarantee these aren't top sorted.

Reply:
Yes, it's fine. Will delete the commented code in the next commit.